wsdl-core/src/main/java/io/ballerina/wsdl/core/XMLSchemaParser.java

```diff
@@ -0,0 +1,318 @@
+import org.apache.ws.commons.schema.XmlSchemaContentModel;
+import org.apache.ws.commons.schema.XmlSchemaElement;
+import org.apache.ws.commons.schema.XmlSchemaEnumerationFacet;
+import org.apache.ws.commons.schema.XmlSchemaFacet;
+import org.apache.ws.commons.schema.XmlSchemaObjectCollection;
+import org.apache.ws.commons.schema.XmlSchemaParticle;
+import org.apache.ws.commons.schema.XmlSchemaSequence;
+import org.apache.ws.commons.schema.XmlSchemaSimpleContent;
+import org.apache.ws.commons.schema.XmlSchemaSimpleContentExtension;
+import org.apache.ws.commons.schema.XmlSchemaSimpleContentRestriction;
+import org.apache.ws.commons.schema.XmlSchemaSimpleType;
+import org.apache.ws.commons.schema.XmlSchemaSimpleTypeContent;
+import org.apache.ws.commons.schema.XmlSchemaSimpleTypeList;
+import org.apache.ws.commons.schema.XmlSchemaSimpleTypeRestriction;
+import org.apache.ws.commons.schema.XmlSchemaSimpleTypeUnion;
+import org.apache.ws.commons.schema.XmlSchemaType;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class XMLSchemaParser {
+    private final XmlSchemaElement rootElement;
+    private final List<Field> fields;
+    private final Map<XmlSchemaType, Field> visitedTypes = new HashMap<>();
+
+    protected XMLSchemaParser(XmlSchemaElement rootElement) {
+        this.rootElement = rootElement;
+        fields = new ArrayList<>();
+    }
+
+    protected List<Field> parseRootElement() {
+        XmlSchemaType elementType = rootElement.getSchemaType();
+        if (elementType instanceof XmlSchemaSimpleType) {
+            fields.add(processSimpleField(rootElement, (XmlSchemaSimpleType) elementType));
+        } else if (elementType instanceof XmlSchemaComplexType) {
+            fields.add(processComplexField(rootElement, (XmlSchemaComplexType) elementType));
+        }
+        return fields;
+    }
+
+    private BasicField processSimpleField(XmlSchemaElement element, XmlSchemaSimpleType elementType) {
+        String name = element.getName();
+        XmlSchemaSimpleTypeContent typeContent = elementType.getContent();
+        if (typeContent == null) {
+            String type = element.getSchemaTypeName().getLocalPart();
+            BasicField elementBasicField = new BasicField(name, type);
+            elementBasicField.setNullable(element.isNillable());
+            elementBasicField.setRequired(element.getMinOccurs() > 0);
+            elementBasicField.setDefaultValue(element.getDefaultValue());
+            elementBasicField.setArray(element.getMaxOccurs() > 1);
+            return elementBasicField;
+        }
+        if (typeContent instanceof XmlSchemaSimpleTypeList) {
+            // TODO: Implement this later
+        } else if (typeContent instanceof XmlSchemaSimpleTypeRestriction) {
+            XmlSchemaSimpleTypeRestriction restriction = (XmlSchemaSimpleTypeRestriction) typeContent;
```

Comment:
Check for other places as well.
```suggestion
        } else if (typeContent instanceof XmlSchemaSimpleTypeRestriction restriction) {
```